x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/warning/warning_step.tsx

```diff
@@ -140,19 +168,27 @@ export const WarningsFlyoutStep: React.FunctionComponent<WarningsConfirmationFly
       <EuiFlyoutFooter>
         <EuiFlexGroup justifyContent="spaceBetween">
           <EuiFlexItem grow={false}>
-            <EuiButtonEmpty iconType="arrowLeft" onClick={hideWarningsStep} flush="left">
+            <EuiButtonEmpty iconType="arrowLeft" onClick={back} flush="left">
               <FormattedMessage
-                id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.cancelButtonLabel"
+                id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.cancelButtonLabel"
                 defaultMessage="Back"
               />
             </EuiButtonEmpty>
           </EuiFlexItem>
           <EuiFlexItem grow={false}>
-            <EuiButton fill color="primary" onClick={continueReindex} disabled={blockAdvance}>
-              <FormattedMessage
-                id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.continueButtonLabel"
-                defaultMessage="Continue reindexing"
-              />
+            <EuiButton fill color="primary" onClick={confirm} disabled={blockAdvance}>
+              {flow === 'reindex' && (
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.reindex.startButtonLabel"
+                  defaultMessage="Start reindexing"
+                />
+              )}
+              {flow === 'readonly' && (
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.readonly.startButtonLabel"
+                  defaultMessage="Make read only"
```

Comment:
In other places we say: "Mark as read only", we should pick one: "Make" or "Mark" and use that everywhere.

Reply:
Going for `Mark as read-only` everywhere (using [hyphen](https://www.reddit.com/r/grammar/comments/svoyjg/comment/hxh8ozw/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) to avoid confusion)